spec/policy/default_credentials/default_credentials_spec.lua

```diff
@@ -0,0 +1,255 @@
+local ipairs = ipairs
+
+local DefaultCredentialsPolicy = require('apicast.policy.default_credentials')
+
+describe('Default credentials policy', function()
+  local policy_with_default_user_key
+  local policy_with_default_app_id_and_key
+  local policy_without_default_creds
+  local policy_with_default_app_id
+  local policy_with_default_app_key
+
+  local service_with_user_key_in_req
```

Comment:
Wouldn't it be better for this to be a function? Then you could drop the before_each and just use local variables.

Reply:
Right. Also, the `before_each` seems to imply that everything there is needed for every test, which is not true.

Changed :+1: